src/components/tl/sharp/tl_sharp_context.c

```diff
@@ -17,24 +17,23 @@ static ucc_mpool_ops_t ucc_tl_sharp_req_mpool_ops = {
 
 static int ucc_tl_sharp_oob_barrier(void *arg)
 {
-    ucc_tl_sharp_oob_ctx_t *oob_ctx = (ucc_tl_sharp_oob_ctx_t *)arg;
-    ucc_tl_sharp_context_t *ctx     = oob_ctx->ctx;
-    ucc_oob_coll_t         *oob_coll;
+    ucc_tl_sharp_oob_ctx_t *oob_ctx  = (ucc_tl_sharp_oob_ctx_t *)arg;
+    ucc_tl_sharp_context_t *ctx      = oob_ctx->ctx;
+    ucc_oob_coll_t         *oob_coll = oob_ctx->oob;
     ucc_status_t            status;
     char                    sbuf, *rbuf;
     void                   *req;
 
-    oob_coll = (oob_ctx->ctx_oob != NULL) ? oob_ctx->ctx_oob: oob_ctx->team_oob;
-    rbuf     = ucc_malloc(sizeof(char) * oob_coll->n_oob_eps, "tmp_barrier");
+    rbuf = ucc_malloc(sizeof(char) * oob_coll->n_oob_eps, "tmp_barrier");
     if (!rbuf) {
         tl_error(ctx->super.super.lib,
                  "failed to allocate %zd bytes for tmp barrier array",
                  sizeof(char) * oob_coll->n_oob_eps);
-        return -1;
```

Comment:
Why changed to ucc status? It is internal sharp coll api: it has signature with return "int". I guess sharp coll expects it to return 0 or -1. It does not know about ucs codes

Reply:
0 success. any -ve number is error.